schema.go

```diff
@@ -86,6 +86,11 @@ func (s *Schema) init() {
 	s.lazyNameMapping = sync.OnceValue(func() NameMapping {
 		return createMappingFromSchema(s)
 	})
+
+	// Validate that the schema does not contain duplicate field IDs.
+	if _, err := IndexNameByID(s); err != nil {
+		panic(err)
+	}
```

Comment:
I think we should not have runtime panics that can be caused by user error, we should make the schema init fallible instead and return an err